### ISSUE TITLE
jest -t will create page for all skipped session

### ISSUE THREAD

**Issue by xuanzhaopeng** [playwright-community/jest-playwright#698]
**Describe the bug**
In PlaywrightEnvironment, we init this.global.page in `setup` function,  when I do `jest -t "my test name"`, it will init page for ever skipped files which consumes a lot of time if I have 50 files but only run 1 file.

**To Reproduce**

**Expected behavior**
Is it possible to init everything in beforeAll? I noticed we could use `skipInitialization `, but how could I initialise programmatically in beforeAll?

**Comment by mmarkelov** [comment]
@xuanzhaopeng I think this is duplicate https://github.com/playwright-community/jest-playwright/issues/112
I looked through this issue, but there is some limitation from `jest` and our environment. Unfortunately we couldn't know the status of current test, during the setup of **playwright** environment

**Comment by mmarkelov** [comment]
@xuanzhaopeng could you please check **next** version? I hope that i found out the way to skip it

**Comment by andrewgremlich** [comment]
I'm not sure if this would be related. If it isn't, then I can make another issue.

I have a `setupFilesAfterEnv` option filled. In the file I have the following code...

```typescript
  console.log(browser);

  await jestPlaywright.resetContext({
    storageState: JSON.parse(process.env.STORAGE || ""),
  });
```

If I run a command like `jest file-to-be-tested.test.ts`, the `console.log(browser)` does come up with a value.

But if I run a command like 'jest --testNamePattern <pattern>` the console.log(browser)` will come up as `undefined`.

**Comment by mmarkelov** [comment]
@andrewgremlich i'll take a look

**Comment by andrewgremlich** [comment]
On the topic of this same issue, when I do run with a `--testNamePattern` flag activated, I don't seem to be getting multiple browser instances per test.

**Comment by andrewgremlich** [comment]
ahhh, I see what you did @mmarkelov  with the most recent commit for 1.6.0-rc3. I see what I kind find.

**Comment by andrewgremlich** [comment]
Couldn't find anything. There isn't a way to filter out tests that don't match the name pattern?

**Comment by mmarkelov** [comment]
@andrewgremlich could you please try **1.6.0-rc4**. Hope it will help

**Comment by andrewgremlich** [comment]
@mmarkelov , I think that did it! Thanks man! And it keeps extra browsers from popping up too.

**Comment by xuanzhaopeng** [comment]
> @xuanzhaopeng could you please check **next** version? I hope that i found out the way to skip it

I somehow missed your message, looks works fancy! But anyway I will also update it and check it!

Thanks a lot

**Comment by mmarkelov** [comment]
should be fixed in **1.6.0**

**Comment by kawazoe** [comment]
I was just trying to use the `testNamePattern` feature and I hit a snag that I think is related to this change.

All the test that are filtered out crashes trying to access `page` in describes because the variable isn't defined. I'm just trying to pass it to a page object, I don't do anything special with it.

For instance, using a testNamePattern that would skip this test:

```
describe('foobar', () => {
  let po: PageObject;

  beforeAll(async () => {
    po = new PageObject(page);
  });
```

will crash with "page is not defined" when padding it to the PageObject constructor.

I'm on v1.6.1 so definitely after this fix.
